java-frontend/src/main/java/org/sonar/java/se/checks/OptionalGetBeforeIsPresentCheck.java

```diff
@@ -168,6 +172,11 @@ public void visitMethodInvocation(MethodInvocationTree tree) {
         context.addExceptionalYield(peek, programState, "java.util.NoSuchElementException", check);
         reportIssue(tree);
         programState = null;
+      } else if (OPTIONAL_FILTER.matches(tree)) {
+        // filter has one parameter, so optional is next item on stack
+        SymbolicValue optionalSV = programState.peekValue(1);
+        // reuse the same optional - will cause FN as we make filtering a no-op
+        constraintManager.setValueFactory(() -> optionalSV);
```

Comment:
I think we can actually avoid this FN, by creating new optional SV and setting the constraints accordingly